### PR TITLE
Function for uniform global first quadrant array

### PR DESCRIPTION
**Pull request by tim-griesbach**
# Function for uniform global first quadrant array

Proposed changes:
Implement a function to compute the global first quadrant array for a uniform partition. This function is used in the respective places in p4est. This PR is a preparation for the parallel I/O PR since this function is required for the upcoming code.

### CODE REVIEW

**Review by cburstedde — CHANGES_REQUESTED** [review]
Thanks, I know this is tedious, but please use the cut function. I'd like to remove the explicit computation from everywhere except inside the cut function.

File: src/p4est_communication.c [hunk 1]
```diff
@@ -398,16 +398,14 @@ p4est_comm_global_first_quadrant (p4est_gloidx_t global_num_quadrants,
   int                 i;
 
   P4EST_ASSERT (gfq != NULL);
+  P4EST_ASSERT (mpisize >= 0);
```
Comment (cburstedde) [inline src/p4est_communication.c:401]: >= 1 bitte.

**Comment by tim-griesbach** [comment]
I changed the code such that `p4est_partition_cut_uint64` is used inside the loop and I added assertions on `mpisize` and `global_num_quadrants`. I used `p4est_partition_cut_uint64` since it was used in the places where now `p4est_comm_global_first_quadrant` is used. However, we could also use `p4est_partition_cut_gloidx` to save the type cast.

**Comment by cburstedde** [comment]
> I changed the code such that `p4est_partition_cut_uint64` is used inside the loop and I added assertions on `mpisize` and `global_num_quadrants`. I used `p4est_partition_cut_uint64` since it was used in the places where now `p4est_comm_global_first_quadrant` is used. However, we could also use `p4est_partition_cut_gloidx` to save the type cast.

Yes please use the gloidx version.

File: src/p4est_communication.c (hunk 1 above)
Comment (tim-griesbach) [reply src/p4est_communication.c]: > `>= 1` bitte.

Done

**Comment by tim-griesbach** [comment]
> Yes please use the gloidx version.

I pushed the corresponding change.

**Comment by tim-griesbach** [comment]
The CI fails due to:
```
There are not enough slots available in the system to satisfy the 2
slots that were requested by the application:

  /home/runner/work/p4est/p4est/build/test/test_conn_reduce3
```
Given our experiences this should be resolved by a re-run of the Github CI action.

**Review by cburstedde — APPROVED** [review]

**Pull request merged**